ir/value.cpp

```diff
@@ -88,7 +87,32 @@ StateValue NullPointerValue::toSMT(State &s) const {
 }
 
 
-void Input::print(std::ostream &os) const {
+void GlobalVariable::print(ostream &os) const {
+  UNREACHABLE();
+}
+
+StateValue GlobalVariable::toSMT(State &s) const {
+  auto sizeexpr = expr::mkInt(allocsize, 64);
+  expr ptrval;
+  unsigned glbvar_bid;
+  if (s.hasGlobalVarBid(getName(), glbvar_bid))
+    // Use the same block id that is used by src
+    ptrval = s.getMemory().alloc(sizeexpr, align, Memory::GLOBAL, glbvar_bid);
+  else {
+    ptrval = s.getMemory().alloc(sizeexpr, align, Memory::GLOBAL, nullopt,
+                                 &glbvar_bid);
+    s.addGlobalVarBid(getName(), glbvar_bid);
+  }
+  if (isconst && initval) {
```

Comment:
is there any reason to store the init val if !isconst?  if not, we shouldn't store it at all.

Reply:
Nope, if a global variable is not constant, it is not necessary to store its value because a caller may change that. This if block states that it is already ignoring the initial value.

Reply:
Then let's remove the `&& initval` bit.

Reply:
Const global may not have its initial value if it is extern variable, so initial is still needed. Instead, I'll leave initval only.